tools/check_envoy_includes.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+# Ensure that includes are referenced via external/envoy
+
+import os, sys
+from pathlib import Path
+
+def get_inspection_targets_from_dir(dir):
+    result = []
+    result.extend(Path(dir).rglob("source/*.cc"))
+    result.extend(Path(dir).rglob("source/*.h"))
+    result.extend(Path(dir).rglob("include/*.h"))
+    result.extend(Path(dir).rglob("test/*.cc"))
+    result.extend(Path(dir).rglob("test/*.h"))
+    return result
+
+def inspect_line(bazel_output_base, file_path, line):
+    if line.startswith('#include "'):
+        path = line[len("#include"):].strip(' "')
+        found_in_nighthawk_sources = os.path.isfile("source/" + path) or os.path.isfile("include/" + path)
+        
+        if not found_in_nighthawk_sources:
+            alternative_found = False
+            potential_envoy_path = os.path.join(bazel_output_base, "external/envoy/", path)             
+            alternative_found = os.path.isfile(potential_envoy_path)
+
+            if not alternative_found:
+                potential_envoy_path = os.path.join(bazel_output_base, "external/envoy/source", path)
+                alternative_found = os.path.isfile(potential_envoy_path)          
+
+            if alternative_found:
+                sys.stderr.writelines("Bad include in file " + str(file_path) + ": " + path)
+                sys.stderr.writelines(" (Possible fixed path: " + potential_envoy_path[len(bazel_output_base):] + ")\n")
+                return False
+
+    return True
+
+def inspect_file(bazel_output_base, file_path):
+    offending_lines = []
+    with open(str(file_path), encoding='utf-8') as f:
+        lines = f.readlines()
+        offending_lines.extend(l for l in lines if not inspect_line(bazel_output_base, file_path, l.strip()))
+    return file_path, offending_lines, len(offending_lines) == 0 
+
+workspace_root = sys.argv[2]
+bazel_output_base = sys.argv[1]
```

Comment:
Maybe make the CLI arg and help print robust.

Reply:
We no longer have any arguments..